etna/transforms/timestamp/holiday.py

```diff
@@ -145,15 +148,78 @@ def fit(self, ts: TSDataset) -> "HolidayTransform":
         -------
         :
             The fitted transform instance.
+
+        Raises
+        ------
+        ValueError:
+            if index timestamp is integer and ``in_column`` isn't set
+        ValueError:
+            if external timestamp isn't datetime
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have frequency
+        ValueError
+            if in ``days_count`` mode external timestamp doesn't have the same frequency for all segments
         """
         if self.in_column is None:
+            if self._mode is HolidayTransformMode.days_count:
+                if ts.freq is None:
+                    raise ValueError("Transform can't work with integer index, parameter in_column should be set!")
+                self._freq = ts.freq
+            else:
+                # set some value that doesn't really matter
+                self._freq = object()  # type: ignore
             self.in_column_regressor = True
         else:
             self.in_column_regressor = self.in_column in ts.regressors
-        self._freq = ts.freq
         super().fit(ts)
         return self
 
+    def _validate_external_timestamps(self, df: pd.DataFrame):
+        df = df.droplevel("feature", axis=1)
+
+        # here we are assuming that every segment has the same timestamp dtype
+        timestamp_dtype = df.dtypes.iloc[0]
+        if not pd.api.types.is_datetime64_dtype(timestamp_dtype):
+            raise ValueError("Transform can work only with datetime external timestamp!")
+
+        if self._mode is HolidayTransformMode.binary or self._mode is HolidayTransformMode.category:
+            return
+
+        segments = df.columns.unique()
+        freq_values = set()
+        for segment in segments:
+            timestamps = df[segment]
+            timestamps = timestamps.loc[timestamps.first_valid_index() :]
+            if len(timestamps) >= 3:
+                cur_freq = pd.infer_freq(timestamps)
+                if cur_freq is None:
+                    raise ValueError(
+                        f"Invalid in_column values! Datetime values should be regular timestamps with some frequency. "
+                        f"This doesn't hold for segment {segment}"
+                    )
+                freq_values.add(cur_freq)
+
+        if len(freq_values) > 1:
+            raise ValueError(
+                f"Invalid in_column values! Datetime values should have the same frequency for every segment. "
+                f"Discovered frequencies: {freq_values}"
+            )
+
+    def _infer_external_freq(self, df: pd.DataFrame) -> str:
+        df = df.droplevel("feature", axis=1)
+
+        # here we are assuming that every segment has the same timestamp dtype
+        timestamp_dtype = df.dtypes.iloc[0]
+        if not pd.api.types.is_datetime64_dtype(timestamp_dtype):
+            raise ValueError("Transform can work only with datetime external timestamp!")
```

Comment:
We have already checked it in `_validate_external_timestamps` method. I think, this lines are redundant.

Reply:
Ok, I'll remove it.